charts/linkerd2-cni/values.yaml

```diff
@@ -41,6 +41,11 @@ privileged: false
 # for more information
 #tolerations:
 
+# -|- NodeAffinity section, See the
+# [K8S documentation](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity)
+# for more information
+#nodeAffinity:
```

Comment:
If you're adding `nodeAffinity` to the linkerd2-cni chart, can you add the corresponding usage in the `cni-plugin.yaml` template file?

Reply:
What would you expect in `cni-plugin.yaml`. I can only see a `DaemonSet` definition. Afaik adding affinity to it doesn't make much sense.

Reply:
What would you expect in `cni-plugin.yaml`. I can only see a `DaemonSet` definition. Afaik adding affinity to it doesn't make much sense.

Reply:
Maybe you have a windows node or some special node you don't like linkerd to run on, and you can express that through affinity rules?

Reply:
Agreed with @alpeb that it might be a good idea to add node affinity to the daemonset. First, it helps us have a consistent set of values that we can tweak, second, it can help to restrict which nodes the daemonset should run (so we don't configure rules for pods running on a specific node -- having a windows node, or something else would make perfect sense in this scenario).

Also, if we're going to add affinity values to the chart we might as well make use of them in the templates. `linkerd-cni` is installed separately, as a cni plugin, these values wouldn't translate to any linkerd template at any point.

Reply:
Good point. Done.